src/syscalls/net.rs

```diff
@@ -265,22 +267,30 @@ extern "C" fn __sys_socket(domain: i32, type_: i32, protocol: i32) -> i32 {
 			let fd = FD_COUNTER.fetch_add(1, Ordering::SeqCst);
 
 			#[cfg(feature = "udp")]
-			if type_ == SOCK_DGRAM {
+			if type_ & SOCK_DGRAM == SOCK_DGRAM {
```

Comment:
Not super relevant for now, but I think it'd be nicer if `type_` was a [_bitflag_](https://docs.rs/bitflags/latest/bitflags/index.html) or something similar.

Reply:
This is the C user space interface. In "high-level" representation I use normally bitflags

Reply:
I revised the code. I use now a bitflag type.